pkg/assets/apiserver/asset_apiserver.go

```diff
@@ -86,6 +85,9 @@ func NewAssetServerConfig(assetConfig oapi.AssetConfig) (*AssetServerConfig, err
 		return nil, err
 	}
 	secureServingOptions := genericapiserveroptions.SecureServingOptions{}
+	secureServingOptions.Listener = listener
+	secureServingOptions.BindAddress = net.ParseIP(host)
```

Comment:
looks good

Reply:
Actually `secureServingOptions.ApplyTo` will not copy BindAddress/Network/Port into the config anymore (there are no fields for that anymore, only the listener). So for `secureServingOptions.ApplyTo` setting them in the options is not needed if the listener is non-nil. But it will not harm, and it seems you rely on those values in the other commit.

Reply:
compare https://github.com/openshift/origin/pull/17576#discussion_r156701169. Does not look that good.